### DISCUSSION TITLE
Add a way to customize the subject title

### DISCUSSION THREAD

**Discussion by mokhosh** [Z3d0X/filament-logger#98] [Ideas]
Hey,
Thanks for the nice package.

I need a way to customize how the subject title is retrieved. Currently this is the hard-coded implementation:

```php
Str::of($state)->afterLast('\\')->headline().' # '.$record->subject_id
```

Currently if you want to override this you will have to copy/paste more than 100 lines of code (the whole `table` method) to change this hard-coded implementation.

I wanted to submit a PR and extract a method which we can override, but I thought I would run the idea by you first, and then do the PR if you're willing to merge.

**Comment by mokhosh** [comment]
This is related https://github.com/Z3d0X/filament-logger/discussions/56

**Comment by Z3d0X** [comment]
implemented in #101 

Here is an example of how this feature will be used
```php
LoggerPlugin::make()
    ->resolveSubjectNameUsing(function (Activity $record, ?string $state) {
        if (is_null($state)) { // No Subject
            return '-';
        }

        return match ($state) {
            Post::class => $record->subject->title,
            User::class => $record->subject->name,
            default => Str::of($state)->afterLast('\\')->headline().' # '.$record->subject_id,
        };
    }),
 ```

**Reply by mokhosh** [reply]
beatiful 🤩
